src/evict/evict_lru.c

```diff
@@ -2150,15 +2150,23 @@ __evict_walk_prepare(WT_SESSION_IMPL *session, uint32_t *walk_flagsp)
     switch (btree->evict_start_type) {
     case WT_EVICT_WALK_NEXT:
         /* Each time when evict_ref is null, alternate between linear and random walk */
-        if (btree->evict_ref == NULL && (++btree->linear_walk_restarts) & 1)
-            /* Alternate with rand_prev so that the start of the tree is visited more often */
-            goto rand_prev;
+        if (btree->evict_ref == NULL && (++btree->linear_walk_restarts) & 1) {
```

Comment:
I understand the code since I’m familiar with the changes made to this part, though it still took me some time to fully grasp it. 

This approach does allow the code to run as it did before the Npos changes. However, I was wondering if it might be helpful to create a ticket to simplify the code further. 

I noticed that the main decision point for where to start in the tree is located [here](https://github.com/wiredtiger/wiredtiger/blame/799f0b68e6d12989f21888d4ae79db8c630f9b1b/src/evict/evict_lru.c#L2084). Do you think it would be beneficial to separate the decision-making logic from the execution path to improve clarity?

Reply:
I'd say it could be a part of `__evict_try_restore_walk_position()` but then we probably don't need `__evict_walk_prepare()`.
There's a lot to fix in Eviction and these things a probably not the highest priority.

Reply:
I will leave that decision to you.

Reply:
lgtm